### PR TITLE
Support for Cocoapods

### PR DESCRIPTION
**Pull request by abbasmousavi**
This PR is supposed to add support for projects using cocoapods for dependency management.

### CODE REVIEW

**Review by mxcl — COMMENTED** [review]
Looks good. Don’t link, that’s up to the final app to do. Variable has a typo. The batterTarget doesn’t need to depend on pods since all its ancestors do. We should get the right target by name; `last` is flakey.

**Comment by abbasmousavi** [comment]
I don't know if it works for multiplatform or multi target projects.

**Comment by mxcl** [comment]
That should be fine. Not relevant here.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift [hunk 1]
```diff
@@ -141,6 +141,24 @@ public class XcakeProject: XcodeProject {
                     try target.depend(on: cakefileTarget)
                 }
             }
+            
+            //// integrate Cocoapods stuff if its there
+            if prefix.Podfile.isFile, prefix.Pods.isDirectory {
+
+                let podPorject = try XcodeProject(existing: prefix.Pods / "Pods.xcodeproj")
+
+                if let podTarget = podPorject.nativeTargets.first(where: { $0.name.contains("Pods-") }) {
```
Comment (mxcl) [inline Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: `hasPrefix` is more robust, theoretically a dependency may be called `FooPods-Bar`.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift [hunk 2]
```diff
@@ -141,6 +141,24 @@ public class XcakeProject: XcodeProject {
                     try target.depend(on: cakefileTarget)
                 }
             }
+            
+            //// integrate Cocoapods stuff if its there
+            if prefix.Podfile.isFile, prefix.Pods.isDirectory {
+
+                let podPorject = try XcodeProject(existing: prefix.Pods / "Pods.xcodeproj")
```
Comment (mxcl) [inline Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: Typo in variable name remains.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift [hunk 3]
```diff
@@ -141,6 +141,24 @@ public class XcakeProject: XcodeProject {
...
+                if let podTarget = podPorject.nativeTargets.first(where: { $0.name.contains("Pods-") }) {
+
+                    for target in batterTargets {
+                        try target.depend(on: podTarget)
+                    }
+
+                    let pods = try mainGroup.add(group: caked, name: .custom("Pods"))
+
+                    self.baseConfiguration = pods.add(file: prefix.Pods / "Target Support Files/\(podTarget.name)/\(podTarget.name).debug.xcconfig", name: .basename)
+                    self.baseConfiguration = pods.add(file: prefix.Pods / "Target Support Files/\(podTarget.name)/\(podTarget.name).release.xcconfig", name: .basename)
```
Comment (mxcl) [inline Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: This is incorrect. You should set the `debug` and `release` base configurations for the batterTargets only. Also, as you might expect, setting the same variable twice means the first setting is ignored.

Comment (gcox) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: Also, configurations aren't limited to "debug" and/or "release", there can be any number of them, named anything.

Comment (mxcl) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: We control this `xcodeproj` though, and we only create debug and release.

Comment (abbasmousavi) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: we are supposing a standard cocoapods installation, that has only release and debug configurations, am I missing something?

Comment (mxcl) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: Here we are modifying `Cake.xcodeproj`, so it doesn’t matter what CocoaPods does.

Comment (abbasmousavi) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: I have fixed it, please take a look.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift [hunk 4]
```diff
@@ -145,18 +145,19 @@ public class XcakeProject: XcodeProject {
             //// integrate Cocoapods stuff if its there
             if prefix.Podfile.isFile, prefix.Pods.isDirectory {
 
-                let podPorject = try XcodeProject(existing: prefix.Pods / "Pods.xcodeproj")
+                let podPoroject = try XcodeProject(existing: prefix.Pods / "Pods.xcodeproj")
```
Comment (mxcl) [inline Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: This is also a typo.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift (hunk 3 above)
Comment (mxcl) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: Looks good.

File: Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift (hunk 4 above)
Comment (abbasmousavi) [reply Sources/Model/Base/XcodeProject/XcakeProject/XcakeProject.swift]: sorry about that!

**Pull request merged**